src/app/policies/views/PolicyTypeListView.vue

```diff
@@ -76,10 +74,11 @@
                                 policyPath: policyType.path,
                               },
                             }"
+                            :mount="route.params.policyPath.length === 0 && i === 0 ? route.replace : undefined"
```

Comment:
> CircuitBreaker used to be the first in the list which is why it was chosen. Now it should be MeshAccessLog. For the Policies tab (or links to the policy list view in general), we can’t link to the first one with non-zero policies because that would mean we’d always have to first evaluate /mesh-insights/:mesh.

https://github.com/kumahq/kuma-gui/issues/2415#issuecomment-2049735176

---

If we wanted to redirect to the first non-zero one we could just look at `meshInsight?.policies?.[policyType.name]?.total` like we do below

I decided to just go to the first one though (whatever that is from now into the future). I'm easy either way, but I kinda think we should just use the first existing even if its zero.

Reply:
Ah also, not totally sure on the name `:mount` here. I started with `:auto` and then tried to use `emit` instead of a prop and `@mount` felt like a better vue-like name, but then you can't detect when the `@mount` function changes I don't think? So I moved it back to `:mount`. If you had any naming suggestions, lmk.

Reply:
Might just call it `replace`. Is there really ever gonna be a different use case for this sort of navigation indirection?

Reply:
Yeah I see what you're thinking,  I was thinking something like `redirect` also.  

Thing is you can put anything in this function not just something that redirects. I can also console.log in there if I wanted to so `:redirect/replace="() => console.log"` doesn't make sense. I think the function means that if there is another use case we haven't thought of, injecting a function/behaviour is a looser coupling/more flexible, and the fact that its a function means "do _something_ on mount".

The other option is that I make it `redirect=true` and then import `useRouter` into `XAction` and redirect from there.

The other thing I was thinking I should avoid is coupling it to RouteView/AppView, although now come to think of it maybe coupling it and doing something similar to RouteTitle would be good. 

I dunno, I'm still a bit in two minds 🤔 I think if you're not super strong on anything here we merge this, and possibly tweak after.

Reply:
Gonna resolve this one I think, I didn't notice you'd already approved thanks for that!